File: FastMines_android/src/main/java/fmg/android/app/serializers/AppDataSerializer.java
package fmg.android.app.serializers;

import android.content.SharedPreferences;

import fmg.android.app.model.MosaicActivityBackupData;
import fmg.common.Logger;
import fmg.core.app.serializers.ISerializer;
import fmg.android.app.model.AppData;
import fmg.android.app.presentation.MenuSettings;

/** Application data (de)serializer. For save / restore {@link AppData} */
public class AppDataSerializer implements ISerializer {

    private static final int VERSION = 1;

    private static final String KEY__APP_DATA                                 = "AppData";
    private static final String KEY__APP_DATA__VERSION                        = KEY__APP_DATA + ".version";
    private static final String KEY__APP_DATA__MENU_SETTINGS__SPLIT_PANE_OPEN = MenuSettings.class.getSimpleName() + ".splitPaneOpen";

    /*
    public AppData load(Bundle from) {
        AppData data = new AppData();
        if (from != null) try {
            MosaicInitData mid = new MosaicInitData();
            new MosaicInitDataSerializer().load(from, mid);
            data.setMosaicInitData(mid);
            data.setSplitPaneOpen(from.getBoolean(KEY__MENU_SETTINGS__SPLIT_PANE_OPEN, true));
        } catch(Exception ex) {
            Logger.error("Can not read app settings from Bundle", ex);
            data = new AppData(); // reset
        }
        return data;
    }

    public void save(AppData data, Bundle to) {
        new MosaicInitDataSerializer().save(data.getMosaicInitData(), to);
        to.putBoolean(KEY__MENU_SETTINGS__SPLIT_PANE_OPEN, data.isSplitPaneOpen());
    }
    */

    public AppData load(SharedPreferences from) {
        AppData data = new AppData();
        if (from != null) try {
            int version = from.getInt(KEY__APP_DATA__VERSION, 1);
            if (version != VERSION)
                throw new IllegalArgumentException("AppDataSerializer: Version #" + version + " is not supported");

            MosaicActivityBackupData mosaicActivityBackupData = new MosaicActivityBackupDataSerializer().load(from);
            if (mosaicActivityBackupData == null)
                data.setMosaicInitData(new MosaicInitDataSerializer().load(from));
            else
                data.setMosaicActivityBackupData(mosaicActivityBackupData);

            data.setSplitPaneOpen(from.getBoolean(KEY__APP_DATA__MENU_SETTINGS__SPLIT_PANE_OPEN, MenuSettings.DEFAULT_SPLIT_PANE_OPEN));
        } catch(Exception ex) {
            Logger.error("Can not read app settings from SharedPreferences", ex);
            data = new AppData(); // reset
        }
        return data;
    }

    public void save(AppData data, SharedPreferences to) {
        SharedPreferences.Editor editor = to.edit();
        editor.putInt(KEY__APP_DATA__VERSION, VERSION);

        MosaicActivityBackupData mosaicActivityBackupData = data.getMosaicActivityBackupData();
        if (mosaicActivityBackupData == null)
            new MosaicInitDataSerializer().save(data.getMosaicInitData(), editor);
        else
            new MosaicActivityBackupDataSerializer().save(mosaicActivityBackupData, editor);

        editor.putBoolean(KEY__APP_DATA__MENU_SETTINGS__SPLIT_PANE_OPEN, data.isSplitPaneOpen());
        editor.apply();
    }

}

File: FastMines_uwp/src/fmg/uwp/app/Serializers/AppDataSerializer.cs
using System;
using Windows.Foundation.Collections;
using Fmg.Common;
using Fmg.Core.App.Model;
using Fmg.Core.App.Serializers;
using Fmg.Uwp.App.Model;
using Fmg.Uwp.App.Presentation;

namespace Fmg.Uwp.App.Serializers {

    /// <summary> Application data (de)serializer. For save / restore <see cref="AppData"/> </summary>
    public class AppDataSerializer : ISerializer {

        /// <summary> Singleton </summary>
        public static MosaicInitData MosaicInitData { get; } = new MosaicInitData();
        /// <summary> Singleton </summary>
        public static MenuSettings MenuSettings { get; } = new MenuSettings();

        private const int VERSION = 1;

        private const string KEY__APP_DATA__VERSION              = "Version";
        private const string KEY__MENU_SETTINGS__SECTION         = nameof(MenuSettings);
        private const string KEY__MENU_SETTINGS__SPLIT_PANE_OPEN = nameof(MenuSettings.SplitPaneOpen);

        public AppData Load(IPropertySet from) {
            try {
                if (!from.ContainsKey(KEY__APP_DATA__VERSION))
                    return new AppData();

                int version = (int)from[KEY__APP_DATA__VERSION];
                if (version != VERSION)
                    throw new ArgumentException($"AppDataSerializer: Version #{version} is not supported");

                AppData data = new AppData();
                var mosaicPageBackupData = new MosaicPageBackupDataSerializer().Load(from);
                if (mosaicPageBackupData == null)
                    data.MosaicInitData = new MosaicInitDataSerializer().Load(from);
                else
                    data.MosaicPageBackupData = mosaicPageBackupData;

                var menuSection = (Windows.Storage.ApplicationDataCompositeValue)from[KEY__MENU_SETTINGS__SECTION];
                data.SplitPaneOpen = (bool)menuSection[KEY__MENU_SETTINGS__SPLIT_PANE_OPEN];

                return data;
            } catch(Exception ex) {
                Logger.Error($"Can`t load app settings: {ex.Message}");
            }
            return new AppData();
        }

        public void Save(AppData data, IPropertySet to) {
            to.Clear();

            to[KEY__APP_DATA__VERSION] = VERSION;

            var mosaicPageBackupData = data.MosaicPageBackupData;
            if (mosaicPageBackupData == null)
                new MosaicInitDataSerializer().Save(data.MosaicInitData, to);
            else
                new MosaicPageBackupDataSerializer().Save(mosaicPageBackupData, to);

            to[KEY__MENU_SETTINGS__SECTION] = new Windows.Storage.ApplicationDataCompositeValue() {
                [KEY__MENU_SETTINGS__SPLIT_PANE_OPEN] = data.SplitPaneOpen,
            };
        }

    }

}
